sql/log_event.cc

```diff
@@ -838,7 +838,7 @@ int Log_event::read_log_event(IO_CACHE* file, String* packet,
       DBUG_RETURN(LOG_READ_MEM);
     memcpy(newpkt, packet->ptr(), ev_offset);
 
-    uint dstlen;
+    uint dstlen= (uint) sz;
```

Comment:
this is why you got failures on `dst[*dstlen - 1]` assert.

here you should have `dstlen= sz - ev_offset - 4`